baize/datastructures.py

```diff
@@ -479,7 +479,7 @@ def __init__(
         super().__init__(_items)
 
         if raw is not None:
-            self._list = raw
+            self._list = list(raw)
```

Comment:
The reason for not doing shallow copy here is to synchronize the modification to raw. But this is the design of starlette, and I am considering refactoring it.

Reply:
The line ensures `self._list` become a Sequence from the iterable `raw` otherwise mypy will cry for that.

if that is the case, `raw` must be a Sequence at least to make the type checker happy.

Reply:
Also, rely on the synchronization side effect is not a good design. It is not explicit and users(developers) can be suprised.